Xamarin.Forms.Platform.Android/AppCompat/NavigationPageRenderer.cs

```diff
@@ -684,6 +681,20 @@ Task<bool> SwitchContentAsync(Page view, bool animated, bool removed = false, bo
 			return tcs.Task;
 		}
 
+		Fragment GetFragment(Page page, bool removed, bool popToRoot)
+		{
+			// pop
+			if (removed)
+				return _fragmentStack[_fragmentStack.Count - 2];
```

Comment:
Negative index would mean we're trying to pop the root and I don't think that's allowed. If we ever end up with a negative index, it's better to let it throw an exception so the issue can be fixed somewhere else.